### PR TITLE
Allow passing USPS package codes

### PR DESCRIPTION
**Pull request by mamhoff**
USPS offers discounts for their special package codes. This PR allows us to pass those package codes as `usps_package_code` properties to the packages container object.

### CODE REVIEW

File: lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb [hunk 1]
```diff
@@ -39,12 +39,16 @@ def serialize_address(address)
...
         def serialize_packages(packages)
           packages.map do |package|
-            {
+            package_hash = {
               weight: {
                 value: package.weight.convert_to(:ounce).value.to_f,
                 unit: "ounce"
               }
             }
+            if package.container.properties[:usps_package_code]
```
[the remark this answers is not in the document]
Comment (mamhoff) [reply lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb]: `ShipEngine` is problematic that way, as it's really a wrapper for a multitude of shipping services, but accepts some options specific to USPS (and others), like the package type. That's why I prefixed the `properties` key with `usps`. For now, I vote for keeping it this way until we (or someone else) runs into a case where splitting things up into multiple classes for different carriers becomes necessary.

File: lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb [hunk 2]
```diff
@@ -45,8 +45,19 @@ def serialize_packages(packages)
                 unit: "ounce"
               }
             }
-            if package.container.properties[:usps_package_code]
-              package_hash.merge!(package_code: package.container.properties[:usps_package_code])
+
+            package_code = package.container.properties[:usps_package_code] || "package"
+            package_hash.merge!(package_code: package_code)
+
+            if package_code == 'package'
```
Comment (tvdeyen) [inline lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb]: Maybe a constant for this magic string would be beneficial?

File: lib/friendly_shipping/services/ship_engine/bad_request.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          @rest_error = rest_error
+          @response = rest_error.response
+          super parse_json_errors
+        end
+
+        private
+
+        def parse_json_errors
+          parsed_body = JSON.parse(response.body)
+          messages = parsed_body['errors'].map { |e| e['message'] }
```
Comment (tvdeyen) [inline lib/friendly_shipping/services/ship_engine/bad_request.rb]: Potential `Undefined method .map for NilClass` here. As we parse this in a error object itself, we must be sure that this code is very safe. Maybe dealing with `JSON::ParseError`  as well?

File: lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb (hunk 2 above)
Comment (dividedharmony) [inline lib/friendly_shipping/services/ship_engine/serialize_label_shipment.rb]: @mamhoff Could `: usps_package_code` actually have the value "package" or is that just a default value?

**Pull request merged**